Automattic/wp-calypso
# Update: convert form phone media input to typescript

#### Changes proposed in this Pull Request

* Somehow the country code passing through this component ended up being undefined in production. I'm still not sure how that happened, but in the course of investigating converted this component to TS.

#### Testing instructions

There should be no observable changes in behavior. This component is used in the country-specific payment method part of checkout. To see it:

- Set your currency to BRL
- Enter checkout with a plan. At the contact details step, set your country to Brazil; the postal code format is 8 digits (like 11111-111)
- At the payment method step, choose the new card method. This will display the extra fields, including the newly converted phone input.
